collector/asp_net_application.go

```diff
@@ -0,0 +1,160 @@
+			nil,
+		),
+		RequestsExecuting: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "requests_executing"),
+			"The number of requests currently executing.",
+			nil,
+			nil,
+		),
+		RequestsInApplicationQueue: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "requests_in_application_queue"),
+			"The number of requests currently in the queue.",
+			nil,
+			nil,
+		),
+		RequestsPerSec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "requests_per_sec"),
+			"The number of requests executed per second.",
+			nil,
+			nil,
+		),
+		RequestWaitTime: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "request_wait_time_ms"),
+			"The number of milliseconds the most recent request waited in the queue before being processed",
+			nil,
+			nil,
+		),
+	}, nil
+}
+
+// Collect sends the metric values for each metric
+// to the provided prometheus Metric channel.
+func (c *asp_net_applicationCollector) Collect(ctx *ScrapeContext, ch chan<- prometheus.Metric) error {
+	if desc, err := c.collect(ch); err != nil {
+		log.Error("failed collecting win32_perfrawdata_aspnet_aspnetapplications metrics:", desc, err)
+		return err
+	}
+	return nil
+}
+
+// Win32_PerfRawData_ASPNET_ASPNETApplications docs:
+// - https://learn.microsoft.com/en-us/previous-versions/aspnet/fxk122b4(v=vs.100)#aspnet-application-performance-counters
+type Win32_PerfRawData_ASPNET_ASPNETApplications struct {
+	Name string
+
+	ErrorsDuringExecution                uint32
+	ErrorsTotalPerSec                    uint32
+	ManagedMemoryUsedestimated           uint32
+	PercentManagedProcessorTimeestimated uint32
+	RequestsExecuting                    uint32
+	RequestsInApplicationQueue           uint32
+	RequestsPerSec                       uint32
+	RequestWaitTime                      uint32
+}
+
+func (c *asp_net_applicationCollector) collect(ch chan<- prometheus.Metric) (*prometheus.Desc, error) {
+	var dst []Win32_PerfRawData_ASPNET_ASPNETApplications
+	q := queryAll(&dst)
+	if err := wmi.Query(q, &dst); err != nil {
+		return nil, err
+	}
+
+	ch <- prometheus.MustNewConstMetric(
+		c.ErrorsDuringExecution,
+		prometheus.GaugeValue,
+		float64(dst[0].ErrorsDuringExecution),
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		c.ErrorsTotalPerSec,
+		prometheus.CounterValue,
+		float64(dst[0].ErrorsTotalPerSec),
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		c.ManagedMemoryUsedestimated,
+		prometheus.GaugeValue,
+		float64(dst[0].ManagedMemoryUsedestimated),
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		c.PercentManagedProcessorTimeestimated,
+		prometheus.GaugeValue,
+		float64(dst[0].PercentManagedProcessorTimeestimated),
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		c.RequestsExecuting,
+		prometheus.GaugeValue,
+		float64(dst[0].RequestsExecuting),
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		c.RequestsInApplicationQueue,
+		prometheus.CounterValue,
+		float64(dst[0].RequestsInApplicationQueue),
+	)
```

Comment:
Should the metric type be Counter? The metric description indicates it's the number of requests currently in the queue, which implies a Gauge metric.